### ISSUE TITLE
[AppImage] Building fails after adding package ingredients

### ISSUE THREAD

**Issue by geigi** [openSUSE/open-build-service#4160]
Issue/Feature description
====================
I'm trying to build an AppImage using OBS for a GTK application written in Python that contains all the dependencies. I've created a working `appimage.yml`, but as soon as I add the dependencies to the yml, the build fails. It's not showing as failed in the Overview, but it also doesn't create a working AppImage.

Expected result
============
Generated AppImage that contains the specified dependencies.

How to Reproduce
===============
1. Try building the `appimage.yml` without the ingredients section. Note that the build is successful.
2. Now add the desired ingredients to the yml. Now the AppImage build fails.

Further information
===============
* Package on OpenSuSE build service: https://build.opensuse.org/package/show/home:geigi:branches:OBS:AppImage:Templates/Cozy
* `appimage.yml`:
```
app: Cozy
binpatch: true

build:
  packages:
    - python3
    - python3-pip
    - python3-setuptools
    - meson
    - linuxdeployqt
    - python3-gst
    - gtk3
    - gtk3-devel
    - glib2
    - glib2-devel
  git:
    - https://github.com/geigi/cozy.git

ingredients:
  packages:
    - gstreamer
    - gstreamer-plugins-base
    - python3
    - gtk3
    - python-mutagen

script:
  - cd $BUILD_SOURCE_DIR/cozy*
  - mkdir bin
  - meson --prefix=$BUILD_APPDIR/usr ./bin
  - ninja -C $BUILD_SOURCE_DIR/cozy/bin install
  - unset QTDIR; unset QT_PLUGIN_PATH ; unset LD_LIBRARY_PATH
  - cd $BUILD_APPDIR/
  - find .
  - cp usr/share/applications/com.github.geigi.cozy.desktop .
  - cp usr/share/icons/hicolor/scalable/apps/com.github.geigi.cozy.svg 
```
- Error in the build log:
```
[   91s] now finalizing build dir...
[   91s] SIOCSIFADDR: File exists
[   91s] Unpacking cozy.obscpio ...
[   91s] 1895 blocks
[   91s] Running build time source services...
[   91s] -----------------------------------------------------------------
[   91s] ----- building appimage.yml (user abuild)
[   91s] -----------------------------------------------------------------
[   91s] -----------------------------------------------------------------
[   91s] creating repository metadata...
[   95s] Spawning worker 0 with 102 pkgs
[   95s] Spawning worker 1 with 102 pkgs
[   95s] Spawning worker 2 with 102 pkgs
[   95s] Spawning worker 3 with 102 pkgs
[   95s] Workers Finished
[   95s] Saving Primary metadata
[   95s] Saving file lists metadata
[   95s] Saving other metadata
[   95s] rpm: no arguments given for query
[   95s] ERROR: Can not find a unique desktop file for  defined in App: tag
[   95s] 
[   95s] cloud109 failed "build appimage.yml" at Fri Nov 24 12:54:37 UTC 2017.
```

**Comment by adrianschroeter** [comment]
There is an error message which could be a bit better, but is true. You don't have a desktop file which is named Cozy as you specified in your appimage.yml.

this is neither an OBS bug, nor a problem in the appimage toolchain.

**Comment by adrianschroeter** [comment]
Note: it is not enough to copy a desktop file manualy when using ingredients, it needs to be provided by the packages. I have improved the error message now.

**Comment by adrianschroeter** [comment]
hm, thinking again about it, there is also a way to support this kind of mixture with base ingredient packages...

Submitted some code changes for that, so it should become to work in some minutes.

**Comment by geigi** [comment]
Thanks Adrian,
I started a fresh project and fixed the desktop file naming issue. The AppImage is now building fine without any error as I can tell, but the ingredients are not included in the resulting AppImage, only the compiled source that is pulled from git.

I looked at some other AppImage projects on OBS and couldn't find a mistake in my setup, but I also didn't find a mixed package like mine.

The project is now here: https://build.opensuse.org/package/show/home:geigi/Cozy
Build log: https://build.opensuse.org/build/home:geigi/AppImage.42.3/x86_64/Cozy/_log
appimage.yml: https://build.opensuse.org/package/view_file/home:geigi/Cozy/appimage.yml?expand=1
